virtool/caches/db.py

```diff
@@ -146,8 +146,12 @@ async def create(
 
         return virtool.utils.base_processor(document)
 
-    except pymongo.errors.DuplicateKeyError:
+    except pymongo.errors.DuplicateKeyError as e:
+        # Check if key-sample.id uniqueness was enforced
         # Keep trying to add the cache with new ids if the generated id is not unique.
+        if "key" in e.details["keyPattern"]:
+            return None
```

Comment:
If I move this then I don't think I'd be able to recursively call `create` to regenerate a `cache_id`, unless you think that is okay to drop

Reply:
You can do like this (not exact):

```py
except pymongo.errors.DuplicateKeyError as e:
  # Keep trying to add the cache with new ids if the generated id is not unique.
  if "_id" in e.details["keyPattern"]:
    recursively_create_until_unique_id()

  raise

I'm not positive the `_id` check is correct. The goal though is to get a unique `_id` for a specific `DuplicateKeyError`. Otherwise the exception is re-raised with `raise` and can be caught in the calling scope.